test/nsswitch_test.rb

```diff
@@ -0,0 +1,125 @@
+#!/usr/bin/env rspec
+# Copyright (c) [2020] SUSE LLC
+#
+# All Rights Reserved.
+#
+# This program is free software; you can redistribute it and/or modify it
+# under the terms of version 2 of the GNU General Public License as published
+# by the Free Software Foundation.
+#
+# This program is distributed in the hope that it will be useful, but WITHOUT
+# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+# more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program; if not, contact SUSE LLC.
+#
+# To contact SUSE LLC about this file by physical or electronic mail, you may
+# find current contact information at www.suse.com.
+
+require_relative "test_helper"
+require "tmpdir"
+
+Yast.import "Nsswitch"
+
+describe Yast::Nsswitch do
+  subject(:nsswitch) { Yast::Nsswitch }
+
+  let(:tmpdir) { Dir.mktmpdir }
+  let(:original_data_example) { File.join(DATA_PATH, "nsswitch.conf/custom/etc") }
+  let(:file_path) { "#{tmpdir}/etc/nsswitch.conf" }
+
+  around do |example|
+    begin
+      FileUtils.cp_r(original_data_example, tmpdir)
+      change_scr_root(tmpdir, &example)
+    ensure
+      FileUtils.remove_entry(tmpdir)
+    end
+  end
```

Comment:
@jreidinger, @imobachgs 

I need help here because at this point the `Nsswitch` module was already required (line 24) and the [`CFA::Nsswitch` loaded](https://github.com/yast/yast-pam/pull/20/files#diff-274c1a3b55f238ac04e89a05f6e8252aR38)

So, the `change_scr_root` has been applied AFTER the `CFA::Nsswitch.load` and test fails because it is looking for the configuration files in the real root.

What can we do here?

Reply:
You can try to memoize the cfa_model (instead of using an attribute).

```ruby
def cfa_model
  @cfa_model ||= CFA::Nsswitch.load
end
```

Reply:
:man_facepalming: :man_facepalming: :man_facepalming: 

So simple! I was obfuscated with the `Yast.import` being called before changing the root and... you know :wink: 

Thanks!